src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp

```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/indexMan/CreateEdgeIndexProcessor.h"
+
+namespace nebula {
+namespace meta {
+
+void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
+    auto spaceID = req.get_space_id();
+    auto indexName = req.get_index_name();
+    CHECK_SPACE_ID_AND_RETURN(spaceID);
+    auto properties = req.get_properties();
+    if (properties.get_fields().size() == 0) {
+        LOG(ERROR) << "Edge's Field should not empty";
+        resp_.set_code(cpp2::ErrorCode::E_INVALID_PARM);
+        onFinished();
+        return;
+    }
+
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::edgeIndexLock());
+    auto ret = getEdgeIndexID(spaceID, indexName);
+    if (ret.ok()) {
+        LOG(ERROR) << "Create Edge Index Failed: " << indexName << " have existed";
+        resp_.set_code(cpp2::ErrorCode::E_EXISTED);
+        onFinished();
+        return;
+    }
+
+    for (auto const &element : properties.get_fields()) {
+        auto edgeName = element.first;
+        auto edgeType = getEdgeType(spaceID, edgeName);
+        if (!edgeType.ok()) {
+            LOG(ERROR) << "Create Edge Index Failed: " << edgeName << " not exist";
+            resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+            onFinished();
+            return;
+        }
+
+        auto fieldsResult = getLatestEdgeFields(spaceID, edgeName);
+        if (!fieldsResult.ok()) {
+            LOG(ERROR) << "Get Latest Edge Property Name Failed";
+            resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+            onFinished();
+            return;
+        }
+        auto fields = fieldsResult.value();
+        for (auto &field : element.second) {
+            if (std::find(fields.begin(), fields.end(), field) == fields.end()) {
```

Comment:
Is better `if (std::count(fields.begin(), fields.end(), field) != 1) `?

Reply:
the recommended is `find`, maybe when the value is found the method will return？

Reply:
I think both are good, just the difference in coding style. 😆